scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala

```diff
@@ -402,10 +401,42 @@ final case class BigQueryTypedTable[T: Coder](
         }
       )
       .pipe(w => Option(params.failedInsertRetryPolicy).fold(w)(w.withFailedInsertRetryPolicy))
-      .pipe(params.configOverride)
-      .pipe(w => if (params.extendedErrorInfo == Disabled) w else w.withExtendedErrorInfo())
+      .pipe(w => Option(params.configOverride).fold(w)(_(w)))
+      .pipe(w => if (params.extendedErrorInfo) w.withExtendedErrorInfo() else w)
+
+    val wr = data.applyInternal(transform)
+
+    var tuple = PCollectionTuple.empty(data.context.pipeline)
+    if (
+      params.method == WriteMethod.FILE_LOADS ||
+      (params.method == WriteMethod.DEFAULT && data.internal.isBounded == PCollection.IsBounded.BOUNDED)
+    ) {
+      tuple = tuple.and(BigQueryIO.SuccessfulTableLoads.tupleTag, wr.getSuccessfulTableLoads)
+    } else if (
+      params.method == WriteMethod.STREAMING_INSERTS ||
+      (params.method == WriteMethod.DEFAULT && data.internal.isBounded == PCollection.IsBounded.UNBOUNDED)
+    ) {
+      tuple = tuple.and(BigQueryIO.SuccessfulInserts.tupleTag, wr.getSuccessfulInserts)
+    } else if (BigQueryIO.storageWriteMethod.contains(params.method)) {
+      tuple = tuple.and(
+        BigQueryIO.SuccessfulStorageApiInserts.tupleTag,
+        wr.getSuccessfulStorageApiInserts
+      )
+    } else {
+      throw new IllegalArgumentException(s"Unknown write method: ${params.method}")
```

Comment:
imo, this seems like more of a case of an `UnsupportedOperationException` than illegal/unknown argument exception